src/analyses/baseUtil.ml

```diff
@@ -0,0 +1,19 @@
+open Prelude.Ana
+open GobConfig
+module Q = Queries
+
+let is_global (a: Q.ask) (v: varinfo): bool =
+  v.vglob || match a (Q.MayEscape v) with `MayBool tv -> tv | _ -> false
+
+let is_static (v:varinfo): bool = v.vstorage == Static
+
+let is_always_unknown variable = variable.vstorage = Extern || Ciltools.is_volatile_tp variable.vtype
```

Comment:
Why are we using physical equality for `vstorage` once and structural equality the other time?

Reply:
No reason I guess, I just extracted those form the current top of `base.ml`. Given that storage is just a basic enum, they should be equivalent in compiled code, so I suppose structural equality should be fine for both and consistency.